frontend/src/components/auth/Register.jsx

```diff
@@ -0,0 +1,524 @@
+import React from "react";
+import {
+  Container,
+  Row,
+  Col,
+  FormGroup,
+  Label,
+  Spinner,
+  Input,
+  InputGroup,
+  Button,
+  InputGroupText,
+} from "reactstrap";
+import { Form, Formik } from "formik";
+import useTitle from "react-use/lib/useTitle";
+
+import { ContentSection, Select } from "@certego/certego-ui";
+
+import { PUBLIC_URL, RECAPTCHA_SITEKEY } from "../../constants/environment";
+import { HACKER_MEME_STRING } from "../../constants";
+import ReCAPTCHAInput from "./utils/ReCAPTCHAInput";
+import {
+  AfterRegistrationModalAlert,
+  InviteOnlyAlert,
+} from "./utils/registration-alert";
+import { registerUser } from "./api";
+import {
+  EmailValidator,
+  PasswordValidator,
+  RecaptchaValidator,
+} from "./utils/validator";
+
+// constants
+const hearAboutUsChoices = [
+  {
+    label: "Other",
+    value: "other",
+  },
+  {
+    label: "Search Engine (Google, DuckDuckGo, etc.)",
+    value: "search_engine",
+  },
+  {
+    label: "Recommended by friend or colleague",
+    value: "was_recommended",
+  },
+  {
+    label: "Social media",
+    value: "social_media",
+  },
+  {
+    label: "Blog or Publication",
+    value: "blog_or_publication",
+  },
+];
+
+const INITIAL_VALUES = {
+  first_name: "",
+  last_name: "",
+  email: "",
+  username: "",
+  password: "",
+  confirmPassword: "",
+  company_name: "",
+  company_role: "",
+  twitter_handle: "",
+  discover_from: "other",
+  recaptcha: "noKey",
+};
+
+const REGISTRATION_FORM_STORAGE_KEY = "registrationForm";
+const initialValues =
+  JSON.parse(localStorage.getItem(REGISTRATION_FORM_STORAGE_KEY, "{}")) ||
+  INITIAL_VALUES;
+
+console.debug("initialValues", initialValues);
+
+const onValidate = (values) => {
+  const errors = {};
+
+  // static text fields
+  const textFields = ["first_name", "last_name", "username"];
+  textFields.forEach((field) => {
+    if (!values[field]) {
+      errors[field] = "Required";
+    } else if (values[field].length > 15) {
+      errors[field] = "Must be 15 characters or less";
+    } else if (values[field].length < 4) {
+      errors[field] = "Must be 4 characters or more";
+    }
+  });
+  if (
+    ["administrator", "admin", "certego", "hacker"].indexOf(values.username) !==
+    -1
+  ) {
+    errors.username = HACKER_MEME_STRING;
+  }
+
+  // email
+  const emailErrors = EmailValidator(values.email);
+  if (emailErrors.email) {
+    errors.email = emailErrors.email;
+  }
+
+  ["company_name", "company_role"].forEach((field) => {
+    if (!values[field]) {
+      errors[field] = "Required";
+    } else if (values[field].length > 30) {
+      errors[field] = "Must be 30 characters or less";
+    } else if (values[field].length < 3) {
+      errors[field] = "Must be 3 characters or more";
+    }
+  });
```

Comment:
I think this can be moved in the validators file, it should be easier to find this logic because are "packaged" with the other validation functions.